src/bootstrap/src/lib.rs

```diff
@@ -655,27 +655,37 @@ impl Build {
     }
 
     /// Gets the space-separated set of activated features for the standard
-    /// library.
+    /// library. This can be configured with the `std_features` key in config.toml.
     fn std_features(&self, target: TargetSelection) -> String {
-        let mut features = " panic-unwind".to_string();
+        let mut features = if let Some(features) = &self.config.rust_std_features {
+            features.iter().map(|s| s.as_ref()).collect::<Vec<&str>>()
+        } else {
+            vec![]
+        };
+
+        if !features.contains(&"panic-unwind") {
+            features.push("panic-unwind");
+        }
```

Comment:
I have two notes for this part:

1. We don't need to use this list as an optional value.
2. Using `BTreeSet` instead of `Vec` for the `rust_std_features` field  would be better, so we don't need to manually check for duplicates.

Reply:
Without an `Option` would this always be required? almost all of the fields inside the `define_config` macro seem to be wrapped in a Option, so I'm trying to understand why this would be different?

Reply:
I am referring to `Config::rust_std_features` not `Rust::std_features`. While parsing the configuration, we can do something like this `config.rust_std_features = std_features.unwrap_or($default_value)`.